Relax tabindex value

axelpale opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. Adding details below will help us to merge your PR faster. -->

### Description

Any tabindex value suffices for an element to be focusable and thus emit keyboard events. The documentation implies that only tabindex value `-1` can do that.

The page in question: https://developer.mozilla.org/en-US/docs/Web/API/Element/keydown_event

### Motivation

The phrase to be improved suggests that keyboard events are **not** emitted when the tabindex attribute of the emitting element has value **other than** `-1`. In reality, based on [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex), [WHATWG](https://html.spec.whatwg.org/multipage/interaction.html#the-tabindex-attribute), and [SO](https://stackoverflow.com/a/6633271/638546), **any** tabindex value makes the element focusable.  All focusable elements receive keyboard events, see [w3c keyboard event order](https://w3c.github.io/uievents/#events-keyboard-event-order). To be sure, I tested that this is the case with Firefox 74 and Chrome 108.

However, the tab key and keyboard events **are** in temporal conflict. The tabindex value `0` and above make the element reachable by pressing the tab key. The special tabindex value `-1` means that the element cannot be navigated to by pressing the tab key. The keyboard events are emitted when a key is pressed, including the tab key. Thus pressing a tab key either a) emits a keyboard event without changing focus, b) changes focus without emitting the keyboard event, or c) emits the keyboard event and then changes focus. It is unclear which of these happens. Therefore the tabindex value `-1` has a special ability to avoid the conflict by not allowing tabbing to reach the keyboard target element. This conflict might have motivated the original author to limit the value to `-1`.

[W3C keyboard event order](https://w3c.github.io/uievents/#events-keyboard-event-order) docs mention this conflict by stating "The event target might change between different key events. For example, a keydown event for the Tab key will likely have a different event target than the keyup event on the same keystroke." This implies that the W3C recommendation is the c) outcome: emit keyboard event and then switch focus.

All in all, any tabindex value will do for element to emit keydown.

### Additional details

<!-- 🔗 Link to release notes, vendor docs, bug trackers, source control, or other places providing more context -->
[MDN tabindex](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex)
[WHATWG tabindex attribute](https://html.spec.whatwg.org/multipage/interaction.html#the-tabindex-attribute)
[W3C keyboard event order](https://w3c.github.io/uievents/#events-keyboard-event-order)


<!-- 👷‍♀️ After submitting, go to the "Checks" tab of your PR for the build status -->

axelpale committed:
Relax tabindex value

The phrase to be improved suggests that keyboard events are **not** emitted when the tabindex attribute of the emitting element has value **other than** `-1`. In reality, based on [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex), [WHATWG](https://html.spec.whatwg.org/multipage/interaction.html#the-tabindex-attribute), and [SO](https://stackoverflow.com/a/6633271/638546), **any** tabindex value makes the element focusable.  All focusable elements receive keyboard events, see [w3c keyboard event order](https://w3c.github.io/uievents/#events-keyboard-event-order). To be sure, I tested that this is the case with Firefox 74 and Chrome 108.

However, the tab key and keyboard events **are** in temporal conflict. The tabindex value `0` and above make the element reachable by pressing the tab key. The special tabindex value `-1` means that the element cannot be navigated to by pressing the tab key. The keyboard events are emitted when a key is pressed, including the tab key. Thus pressing a tab key either a) emits a keyboard event without changing focus, b) changes focus without emitting the keyboard event, or c) emits the keyboard event and then changes focus. It is unclear which of these happens. Therefore the tabindex value `-1` has a special ability to avoid the conflict by not allowing tabbing to reach the keyboard target element. This conflict might have motivated the original author to limit the value to `-1`.

[W3C keyboard event order](https://w3c.github.io/uievents/#events-keyboard-event-order) docs mention this conflict by stating "The event target might change between different key events. For example, a keydown event for the Tab key will likely have a different event target than the keyup event on the same keystroke." This implies that the W3C recommendation is the c) outcome: emit keyboard event and then switch focus.

All in all, any tabindex value will do for element to emit keydown.

estelle commented on files/en-us/web/api/element/keydown_event/index.md:
```diff
@@ -22,7 +22,7 @@ Unlike the deprecated {{domxref("Element/keypress_event", "keypress")}} event, t
 The `keydown` and [`keyup`](/en-US/docs/Web/API/Element/keyup_event) events provide a code indicating which key is pressed, while `keypress` indicates which character was entered. For example, a lowercase "a" will be reported as 65 by `keydown` and `keyup`, but as 97 by `keypress`. An uppercase "A" is reported as 65 by all events.
 
-Keyboard events are only generated by `<inputs>`, `<textarea>` and anything with the `contentEditable` attribute or with `tabindex="-1"`.
+Keyboard events are only generated by `<inputs>`, `<textarea>` and anything with the `contentEditable` or `tabindex` attribute.
```
how about `<summary>`?

estelle reviewed (commented):
`<summary>` also generates keyboard events.

axelpale committed:
Document emitters of keyup event

List elements that emit keyup events in the keyup docs in order to be consistent with the keydown event documentation.

axelpale committed:
Add summary element as possible emitter

Also, correct non-existent `<inputs>` element to `<input>`

axelpale commented:
In order to be consistent, this PR now adds the same sentence to [keyup docs](https://developer.mozilla.org/en-US/docs/Web/API/Element/keyup_event)

The proposed two sentences now both include the `<summary>` element. To support the statement that the summary element is an emitter of keyboard events, see for example this [SO entry](https://stackoverflow.com/q/49974019/638546) that deals with key events that originate from the summary element.

axelpale committed:
Merge branch 'main' into patch-2
